nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,487 @@
+        each seperate region of its corresponding family of brain maps.
+
+    signals_: a numpy array
+        a list of averaged timeseries signals of the subjects extracted from
+        each region.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, maps_img, mask_img=None, target_affine=None,
+                 target_shape=None, standardize=False, low_pass=None,
+                 high_pass=None, t_r=None, memory=Memory(cachedir=None),
+                 min_size=20, thresholding_strategy='percentile',
+                 threshold='auto', extractor='local_regions',
+                 peak_local_smooth=6., verbose=0):
+        self.maps_img = maps_img
+        self.mask_img = mask_img
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.standardize = standardize
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+
+        # parameters for region extraction
+        self.min_size = min_size
+        self.thresholding_strategy = thresholding_strategy
+        self.threshold = threshold
+        self.extractor = extractor
+        self.peak_local_smooth = peak_local_smooth
+
+        self.verbose = verbose
+
+    def fit(self, X=None, y=None):
+        """ Prepare or set up the data for the region extraction
+
+        """
+        maps_img = check_niimg_4d(self.maps_img)
+
+        # Asking NiftiMapsMasker to prepare the data for regions extraction
+        # only if mask_img is provided
+        if self.mask_img is not None:
+            nifti_maps_masker = NiftiMapsMasker(maps_img, mask_img=self.mask_img,
+                                                resampling_target="maps")
+            nifti_maps_masker.fit()
+            self.maps_img_ = nifti_maps_masker.maps_img_
+            self.mask_img_ = nifti_maps_masker.mask_img_
+        else:
+            self.mask_img_ = None
+            self.maps_img_ = maps_img
+
+        # foreground extraction
+        self.threshold_maps_img_ = foreground_extraction(
+            self.maps_img_,
+            mask_img=self.mask_img_,
+            threshold=self.threshold,
+            thresholding_strategy=self.thresholding_strategy)
+
+        # connected component extraction
+        all_regions = []
+        index_each_map = []
+        self.regions_, self.identity_ = connected_component_extraction(
+            self.threshold_maps_img_,
+            self.min_size,
+            self.extractor,
+            self.peak_local_smooth)
+
+        return self
+
+    def fit_transform(self, imgs, confounds=None):
+        return self.fit().transform(imgs, confounds=confounds)
+
+    def transform(self, imgs, confounds=None):
```

Comment:
You don't need this function. If you call `fit` as in my previous comment, you will rely on NiftiMapsMasker.transform and you don't have to do anything.